label_studio/tests/users.tavern.yml

```diff
@@ -83,3 +83,99 @@ stages:
       status_code: 302
       headers:
         location: "/projects/" # redirect to projects page rather than malicious.com
+
+---
+test_name: test_users_soft_delete
+strict: false
+marks:
+- usefixtures:
+  - django_live_url
+stages:
+- id: signup
+  type: ref
+- id: get_my_user
+  type: ref
+- name: get_active_organization
+  request:
+    method: GET
+    url: '{django_live_url}/api/users/{user_pk}'
+  response:
+    status_code: 200
+    save:
+      json:
+        org_pk: active_organization
+- type: ref
+  id: get_invite_url
+- type: ref
+  id: logout
+# signup 2 new users
+- name: signup_new_user_under_active_organization
+  request:
+    url: "{django_live_url}{invite_url}"
+    data:
+      email: test_user@heartextest.com
+      password: 12345678
+    method: POST
+  response:
+    status_code: 302
+- type: ref
+  id: logout
+- name: signup_second_new_user_under_active_organization
+  request:
+    url: "{django_live_url}{invite_url}"
+    data:
+      email: test_second_user@heartextest.com
+      password: 12345678
+    method: POST
+  response:
+    status_code: 302
+- id: get_my_user # get user_pk to use in soft-delete
+  type: ref
+- id: get_user_token
+  type: ref
+- id: logout
+  type: ref
+- id: get_my_user_with_token
+  name: Get my user with token
+  request:
+    headers:
+      authorization: "Token {user_token}"
+    url: "{django_live_url}/api/current-user/whoami"
+    method: GET
+  response:
+    status_code: 200
+- name: login_as_first_new_user
+  request:
+    url: "{django_live_url}/user/login"
+    data:
+      email: test_user@heartextest.com
+      password: 12345678
+    method: POST
+  response:
+    status_code: 302
+- name: soft_delete_user_fails_without_owner_logged_in
+  request:
+    url: "{django_live_url}/users/{user_pk}/soft-delete"
+    method: DELETE
+  response:
+    status_code: 403
+- id: logout
+  type: ref
+- id: login # as owner
+  type: ref
+- name: soft_delete_user_succeeds_as_owner
+  request:
+    url: "{django_live_url}/users/{user_pk}/soft-delete"
+    method: DELETE
+  response:
+    status_code: 204
+- id: logout
+  type: ref
+- name: get_my_user_with_token_fails_as_user_is_now_deleted
+  request:
+    headers:
+      authorization: "Token {user_token}"
+    url: "{django_live_url}/api/current-user/whoami"
+    method: GET
+  response:
+    status_code: 401
```

Comment:
The reason this test passes is that the tokens were deleted, even though `is_deleted` wasn't actually set on the user as far as I can tell (see https://github.com/HumanSignal/label-studio/pull/4876/files#diff-18d3ee828d3b90980ee44c34eae7c6aa18eede8d3024d74149de34ebd7e77beaR203). Is there a more reliable way we could identify whether the user was soft deleted in test, eg a get request for the user by id?

Reply:
Yep, I noticed this as well, I'm now adding methods to check this to the tests:
1. try deleting the user twice, now after the first deletion they should no longer be found by the endpoint and second deletion will fail
2. Try logging in with the user after they have been deleted, it should fail with a 200 (ironically) as 302 indicates successful login and redirection to projects page.